### PR TITLE
Rsyslog capabilitites

### PR DESCRIPTION
**Pull request by sopos**
extending the test with coverage of two additional BZs

### CODE REVIEW

File: rsyslog/Sanity/capabilities-drop/runtest.sh [hunk 1]
```diff
@@ -46,7 +46,7 @@ rlJournalStart && {
     rsyslogConfigAddTo "GLOBALS" < <(rsyslogConfigCreateSection 'CAP')
   rlPhaseEnd; }
 
-  exp_caps="block_suspend chown ipc_lock lease net_admin net_bind_service perfmon setgid setuid sys_admin sys_chroot syslog sys_resource"
+  exp_caps="block_suspend chown dac_override ipc_lock lease net_admin net_bind_service perfmon setgid setuid sys_admin sys_chroot syslog sys_resource"
```
Comment (Cropi) [inline rsyslog/Sanity/capabilities-drop/runtest.sh:49]: Recently we discovered that the CAP_PERFMON capability is not needed and it was removed from upstream as well as part of the original PR. Defined capabilities: https://github.com/rsyslog/rsyslog/blob/master/tools/rsyslogd.c#L2181

**Review by Cropi — CHANGES_REQUESTED** [review]
The PR looks good, just the check for PERFMON capability needs to be removed. Then we can merge it.

File: rsyslog/Sanity/capabilities-drop/runtest.sh (hunk 1 above)
Comment (sopos) [reply rsyslog/Sanity/capabilities-drop/runtest.sh:49]: Is that written somewhere? Like can I base the change on some particular issue?

Comment (Cropi) [reply rsyslog/Sanity/capabilities-drop/runtest.sh:49]: The initial list was mentioned at https://github.com/rsyslog/rsyslog/pull/4999#issuecomment-1313362425 .

Comment (sopos) [reply rsyslog/Sanity/capabilities-drop/runtest.sh:49]: Ok, the list was created based on some of the actual rsyslog builds.

**Comment by Cropi** [comment]
Just one another note regarding changing credentials between zero and non-zero user IDs [1]:
```
To  preserve  the  traditional  semantics  for  transitions between 0 and nonzero user IDs,
the kernel makes some changes to a thread's capability sets on changes to the
thread's real, effective, saved set, and filesystem user IDs (using setuid(2), setresuid(2), or similar).  
...
```
From rsyslog point of view, whenever the user ID is changed to a nonzero UID all the capabilities are dropped. If I understood it correctly then this only happens when changing user ID but not for group ID.

[1] https://man7.org/linux/man-pages/man7/capabilities.7.html

**Comment by sopos** [comment]
> From rsyslog point of view, whenever the user ID is changed to a nonzero UID all the capabilities are dropped. If I understood it correctly then this only happens when changing user ID but not for group ID.

So to remove the ` || "$GROUP" == "root"` part of the condition?

**Comment by Cropi** [comment]
> > From rsyslog point of view, whenever the user ID is changed to a nonzero UID all the capabilities are dropped. If I understood it correctly then this only happens when changing user ID but not for group ID.
> 
> So to remove the ` || "$GROUP" == "root"` part of the condition?

Yes.

Maybe one more thing that should be modified is:
```
65       rlRun -s 'pscap -a | grep rsyslogd'
```
The exit code of this command will be non-zero if user is non-root due to ```grep``` not being able to find the rsyslogd entry.

**Comment by sopos** [comment]
> The exit code of this command will be non-zero if user is non-root due to `grep` not being able to find the rsyslogd entry.

good point

**Review by Cropi — APPROVED** [review]
LGTM!

**Pull request merged**